src/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -0,0 +1,48 @@
+import * as RadixDropdownMenu from '@radix-ui/react-dropdown-menu';
+import classNames from 'classnames';
+import React from 'react';
+
+import styles from './dropdown-menu.module.scss';
+
+export interface DropdownMenuProps extends React.PropsWithChildren {
+    trigger: React.ReactElement;
+    contentProps?: RadixDropdownMenu.DropdownMenuContentProps & React.RefAttributes<HTMLDivElement>;
+}
+
+export const DropdownMenu = ({ trigger, children, contentProps = {} }: DropdownMenuProps) => (
+    <RadixDropdownMenu.Root>
+        <RadixDropdownMenu.Trigger asChild className={styles.trigger}>
+            {trigger}
+        </RadixDropdownMenu.Trigger>
+
+        <RadixDropdownMenu.Portal>
+            <RadixDropdownMenu.Content
+                // Avoid Chrome adding :focus-visible outline to the trigger on close.
+                onCloseAutoFocus={(e) => e.preventDefault()}
+                {...{ collisionPadding: 20, ...contentProps }}
+                className={classNames(styles.content, contentProps.className)}
+            >
+                {children}
+            </RadixDropdownMenu.Content>
+        </RadixDropdownMenu.Portal>
+    </RadixDropdownMenu.Root>
+);
+
+export type DropdownMenuItemProps = RadixDropdownMenu.DropdownMenuItemProps &
+    React.RefAttributes<HTMLDivElement>;
+
+export const DropdownMenuItem = ({ className, ...restProps }: DropdownMenuItemProps) => (
+    <RadixDropdownMenu.Item className={classNames(styles.item, className)} {...restProps} />
+);
+
+export type DropdownMenuSeparatorProps = RadixDropdownMenu.DropdownMenuSeparatorProps &
+    React.RefAttributes<HTMLDivElement>;
```

Comment:
> React.RefAttributes<HTMLDivElement>;

@olehrakwix you've added this, but you're not actually passing a ref? I know in React 19 or something they want to remove `forwardRef`, but we currently use React 18 and it doesn't allow to pass ref via props.